### ISSUE TITLE
vertical space added before and after listing when using bgcolor option

### ISSUE THREAD

**Issue by EWurbel** [gpoore/minted#220]
switching from minted 2.1 to minted 2.5 added vertical space before and after listings only if these listings use the bgcolor option.

example
```
\documentclass{beamer}

\mode<presentation> {
  \usetheme{Warsaw}
  \setbeamercovered{transparent}
}
\usepackage[french]{babel}
% to use beamer translator package for french
\uselanguage{French}
\languagealias{french}{French}

\usepackage[utf8x]{inputenc}

\usepackage[T1]{fontenc}
\usepackage{minted}


\definecolor{bgcode}{rgb}{0.90,0.90,0.90}

\usemintedstyle{tango}
\newminted{c}{fontsize=\tiny,bgcolor=bgcode}

\title{Minted test}



\begin{document}

\begin{frame}
  \titlepage
\end{frame}

\begin{frame}[fragile]{Minted test}
  \begin{itemize}
  \item some text
  \item some text
  \item some text
  \item some text
  \item some text
  \end{itemize}
  \begin{example}
    \begin{minted}[gobble=6]{c}
      int main() 
      { 
        printf("hello, world"); 
        return 0; 
      }
    \end{minted}
  \end{example}
\end{frame}

\begin{frame}[fragile]{Minted test}
  \begin{itemize}
  \item some text
  \item some text
  \item some text
  \item some text
  \item some text
  \end{itemize}
  \begin{example}
    Some text before.
    \begin{ccode*}{gobble=6,fontsize=\scriptsize,texcl}
      int main() 
      { 
        /* \'enonc\'e */
        printf("hello, world"); 
        return 0; 
      }
    \end{ccode*}
    Some text after.
  \end{example}
\end{frame}

\end{document}
```
Result with minted 2.1 (beamer 3.50)
[test-beamer-minted.pdf](https://github.com/gpoore/minted/files/2624043/test-beamer-minted.pdf)

Result with minted 2.5 (beamer 3.36)
[test-beamer-minted.pdf](https://github.com/gpoore/minted/files/2624049/test-beamer-minted.pdf)

The problem is not tied to beamer. It appears also with article class

**Comment by EWurbel** [comment]
Thank you for the answer. I already switched back to minted 2.1 for my impacted documents, using the same solution you suggest, but thanks for the trick. I keep your second hack at hand just in case.

**Comment by moritzmathes** [comment]
I came across the same issue in minted 2.6. The workaround from above didn't work for me. With inspiration from a [LaTeX Stack Exchange article](https://tex.stackexchange.com/questions/228058/how-to-space-before-and-after-a-minted-code-block-with-bgcolor) I came up with the following workaround for the preamble:
```
\AfterEndEnvironment{minted}{\vspace{-1cm}}
```
Perhaps at one point we will get an official fix.

**Comment by vermiculus** [comment]
A slight addendum to @moritzmathes's approach, avoiding absolute lengths and drawing from egreg's fount of genius (the link to which is now lost to my browser history):

```
\AfterEndEnvironment{minted}{\par\vspace{-\baselineskip}}
```

gives me good results:

<img width="493" alt="image" src="https://github.com/gpoore/minted/assets/2082195/86a8f083-0ac7-4116-8b38-4b088aae79ef">

Thanks for your continued work on minted 🙂

**Comment by gpoore** [comment]
This is now fixed in `minted` v3.0, which is available as a [beta on GitHub](https://github.com/gpoore/minted/tree/v3-dev) and will have a final release soon. The `minted` v3.0 beta currently requires the [dev version of `fvextra`](https://github.com/gpoore/fvextra/blob/master/fvextra/fvextra.sty). A new `fvextra` release will be on CTAN within the next few days.
